core/model/src/driver.rs

```diff
@@ -236,3 +236,92 @@ impl RpcMessage for ValidateAllocation {
     type Item = bool;
     type Error = GenericError;
 }
+
+// ************************** ENTER **************************
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Enter {
+    amount: String,
```

Comment:
Parsing amount should be done in CLI, not in driver.
```suggestion
    amount: BigDecimal,
```